packages/suite-data/src/guide/index.ts

```diff
@@ -15,4 +16,13 @@ const fetchSource = async () => {
     await git.checkout(GB_REVISION);
 };
 
-fetchSource();
+const main = async () => {
+    // Fetch content from GitBook mirror.
+    await fetchSource();
+    // Parse content tree.
+    const index = parse();
+
+    console.log(JSON.stringify(index, undefined, 2));
+};
```

Comment:
This part is confusing for me a little bit without digging into code further. 
You fetch something here and you parse something here, but the connection between those two functions calls is not visible.

Reply:
I understand your confusion. The missing link that connects these two functions are the constants in `constants.ts` and the filesystem, or the `TMP` directory to be specific.

I agree that having the global constants is not quite clear and it would be better to pass them explicitly to each of the steps. (fetch, parse, transform) I didn't do that because then I'd have to pass the constants all the way down throu all the functions. That would be lot of repeating arguments.

What about wrapping each of the steps into a class, that accepts the constants as constructor params. We could then have something like

```typescript
// index.ts

const main = () => {
    const fetcher = new Fetcher(GITBOOK_SOURCE, GITBOOK_REVISION, TMP);
    fetcher.fetch();
    
    const parser = new Parser(TMP)
    const index = parser.parse()
    
    const transformer = new Transformer(TMP, DESTINATION)
    transformer.transform()
```

What do you think?

Reply:
That would also make it much easier to test the methods. .)

Reply:
@tomasklim Addressed in https://github.com/trezor/trezor-suite/pull/3811/commits/3809dfc53e0a78a26b9a29f90ed479ebdc2d93dd, https://github.com/trezor/trezor-suite/pull/3811/commits/a8c5235e93dcc57567f4b84fa1e73d16eaeb3ad8 and https://github.com/trezor/trezor-suite/pull/3811/commits/2bb947109c116cb12889e745360e4af6889e6e1d (had to use 3 fixup commits as this improvement is spread over 3 commits.)

I used the class only for the parser. The fetching and transforming are fine with plain old parameters.

Reply:
Definitely better!